src/python_inspector/api.py

```diff
@@ -0,0 +1,373 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# ScanCode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/python-inspector for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+import os
+from typing import Dict
+from typing import List
+from typing import NamedTuple
+from typing import Sequence
+
+from packaging.requirements import Requirement
+from resolvelib import BaseReporter
+from resolvelib import Resolver
+from tinynetrc import Netrc
+
+from _packagedcode.models import DependentPackage
+from _packagedcode.models import PackageData
+from _packagedcode.pypi import PipRequirementsFileHandler
+from _packagedcode.pypi import PythonSetupPyHandler
+from _packagedcode.pypi import can_process_dependent_package
+from python_inspector import DEFAULT_PYTHON_VERSION
+from python_inspector import dependencies
+from python_inspector import utils
+from python_inspector import utils_pypi
+from python_inspector.package_data import get_pypi_data_from_purl
+from python_inspector.resolution import PythonInputProvider
+from python_inspector.resolution import format_pdt_tree
+from python_inspector.resolution import format_resolution
+from python_inspector.resolution import get_deps_from_distribution
+from python_inspector.resolution import get_environment_marker_from_environment
+from python_inspector.resolution import get_package_list
+from python_inspector.resolution import get_python_version_from_env_tag
+from python_inspector.resolution import parse_deps_from_setup_py_insecurely
+from python_inspector.utils import contain_string
+from python_inspector.utils_pypi import PYPI_SIMPLE_URL
+from python_inspector.utils_pypi import Environment
+
+
+class Resolution(NamedTuple):
+    """
+    Return a dependency resolution returned from the ``resolver_api``.
+
+    ``resolution`` is a graph of resolved dependencies.
+    ``packages`` is a list of ``PackageData`` objects.
+    ``files`` is a parsed list of input file data.
+    """
+
+    resolution: Dict
+    packages: List[PackageData]
+    files: List[Dict]
+
+    def to_dict(self):
+        return {
+            "files": self.files,
+            "packages": [package for package in self.packages],
+            "resolution": self.resolution,
+        }
+
+
+def resolver_api(
```

Comment:
This is a better name IMHO:
```suggestion
def resolve_dependencies(
```